packages/rs-drive-abci/src/platform_types/platform_state/v0/mod.rs

```diff
@@ -164,12 +165,17 @@ impl TryFrom<PlatformStateV0> for PlatformStateForSavingV0 {
     }
 }
 
-impl From<PlatformStateForSavingV0> for PlatformStateV0 {
-    fn from(value: PlatformStateForSavingV0) -> Self {
-        PlatformStateV0 {
+impl TryFromPlatformVersioned<PlatformStateForSavingV0> for PlatformStateV0 {
+    type Error = Error;
+
+    fn try_from_platform_versioned(
+        value: PlatformStateForSavingV0,
+        platform_version: &PlatformVersion,
+    ) -> Result<Self, Self::Error> {
+        Ok(PlatformStateV0 {
             genesis_block_info: value.genesis_block_info,
             last_committed_block_info: value.last_committed_block_info,
-            current_protocol_version_in_consensus: value.current_protocol_version_in_consensus,
+            current_protocol_version_in_consensus: platform_version.protocol_version,
```

Comment:
No, this seems wrong. You can't change the current protocol version in consensus.